core/src/test/java/kafka/zk/ReceiptEvent.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.zk;
+
+import org.apache.zookeeper.server.MutedServerCxn;
+import org.apache.zookeeper.server.Request;
+import org.apache.zookeeper.server.ServerCnxn;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * Event corresponding to the invocation of the Zookeeper request processor for a request.
+ */
+public class ReceiptEvent {
+    private final int opCode;
+    private final Lock lock = new ReentrantLock();
+    private final Condition received = lock.newCondition();
+    private final Condition processed = lock.newCondition();
```

Comment:
Sure, I replaced the `Condition` with a completable future and removed the `State` enum altogether as it was unnecessary (since no client code is observing the received event).